src/components/light.js

```diff
@@ -79,9 +68,21 @@ module.exports.Component = registerComponent('light', {
       return;
     }
 
-    // No light yet. Create and add light.
-    this.light = getLight(data);
-    if (this.light) {
+    // No light yet or light type has changed. Create and add light.
+    this.setLight(this.data);
+  },
+
+  setLight: function (data) {
```

Comment:
Ok, I was just thinking that maybe could be nice to refactor that part as the two parts where you recreate the light (if empty or if type changes) does mostly the same